connect/runtime/src/main/java/org/apache/kafka/connect/runtime/distributed/ClusterConfigState.java

```diff
@@ -137,12 +138,13 @@ public TargetState targetState(String connector) {
      * {@link org.apache.kafka.common.config.ConfigTransformer} by having all variable
      * references replaced with the current values from external instances of
      * {@link ConfigProvider}, and may include secrets.
+     * NOTED: WorkerConfigTransformer#transform won't be called if the configuration of task doesn't exist.
```

Comment:
Perhaps replace this NOTE with `If the configuration is not null, it will have been transformed...`